frontend/src/Hooks/useCollapsible.tsx

```diff
@@ -0,0 +1,18 @@
+import { useState } from 'react';
+
+function useCollapsible(initialStates: boolean[]) {
+  const [isOpenArray, setIsOpenArray] = useState(initialStates);
+
+  const toggleCollapse = (index: number) => {
+    const updatesIsOpenArray = [...isOpenArray];
+    updatesIsOpenArray[index] = !updatesIsOpenArray[index];
+    setIsOpenArray(updatesIsOpenArray);
+  };
+
+  return {
```

Comment:
A tip: You probably want this to be inside a `useMemo`, otherwise it will generate a new object on each invocation, and then always rerender